### PR TITLE
fix: Allow deletion of Daily Notes

### PR DESCRIPTION
**Pull request by itsrainingmani**
This PR handles #528 by removing the check for `timeline-page?` in `node_page.cljs`. This renders the "Delete Page" button for Daily Notes.

There might still be some work done on this PR, particularly around the hiding of empty Daily Notes from the All Page List and Search.

### CODE REVIEW

**Comment by tangjeff0** [comment]
> There might still be some work done on this PR, particularly around the hiding of empty Daily Notes from the All Page List and Search.

Can probably do that in a separate PR.

**Comment by itsrainingmani** [comment]
Sounds good 👍🏼

**Comment by itsrainingmani** [comment]
I created a separate `reg-event-fx` to handle the deleting a daily note on the Daily Notes page. This handler takes care of an edge case that I found while testing and removes the item being deleted from the `:daily-notes/items` vector in the db.

File: src/cljs/athens/events.cljs [hunk 1]
```diff
@@ -338,6 +338,18 @@
          :dispatch [:page/create title uid]}))))
 
 
+(reg-event-fx
+  :daily-note/delete
+  (fn [{:keys [db]} [_ uid]]
+    (let [retract-blocks (db/retract-uid-recursively uid)
+          tx-data        (if (db/e-by-av :block/uid uid)  ;; If a page for the Daily Note exists
+                           retract-blocks                 ;; retract both the page and the child blocks
+                           (rest retract-blocks))         ;; else, only retract the child blocks
```
Comment (tangjeff0) [inline src/cljs/athens/events.cljs:347]: Not sure I understand this `if` statement. How could a Daily Note page exist without children? 

Second, why not just dispatch `:page/delete` from within `:daily-note/delete`? All the code from `:page/delete` can be reused. For instance, looking at `:page/delete`, I see there is also a condition that checks for Linked Refs, that is not being checked here.

**Review by tangjeff0 — CHANGES_REQUESTED** [review]

File: src/cljs/athens/events.cljs (hunk 1 above)
Comment (itsrainingmani) [reply src/cljs/athens/events.cljs:347]: I was trying to workaround this issue that popped up when dispatching `:page/delete` on the Daily Notes page. 

`"EXCEPTION" #error {:message "Nothing found for entity id [:block/uid \"02-16-2021\"]", :data {:error :entity-id/missing, :entity-id [:block/uid "02-16-2021"]}}`.

The only way I could get it to work was to remove the `[:db/retractEntity [:block/uid "02-16-2021"]]` item from the `retract-blocks` vector and pass that on to `tx-data`

Good catch on the linked refs part. I've fixed that now.

File: src/cljs/athens/events.cljs [hunk 2]
```diff
@@ -338,6 +338,20 @@
          :dispatch [:page/create title uid]}))))
 
 
+(reg-event-fx
+  :daily-note/delete
+  (fn [{:keys [db]} [_ uid title]]
+    (let [all-blocks         (db/retract-uid-recursively uid)
+          retract-blocks     (if (db/e-by-av :block/uid uid)  ;; If a page for the Daily Note exists
+                               all-blocks                 ;; retract both the page and the child blocks
+                               (rest all-blocks))         ;; else, only retract the child blocks
```
Comment (tangjeff0) [inline src/cljs/athens/events.cljs:347]: I still don't understand how a date page have child blocks if the date page doesn't exist.

Comment (tangjeff0) [reply src/cljs/athens/events.cljs:347]: Why not just dispatch `:page/delete` from `:daily-note/delete`? (Assuming tx-data looks the same)

i.e.

```
(reg-event-fx
  :daily-note/delete
  (fn [{:keys [db]} [_ uid title]]
    (let [filtered-dn        (filterv #(not= % uid) (:daily-notes/items db)) ;; Filter current date from daily note vec
          new-db (assoc db :daily-notes/items filtered-dn)]
      {:fx [[:dispatch [:page/delete uid title]]]
       :db new-db})))
```

Comment (itsrainingmani) [reply src/cljs/athens/events.cljs:347]: Ok. So I figured out why that error was popping up. When I just dispatched the `:page/delete` event directly from the button click. the daily note would be retracted, but it was still in the daily notes vector. The code here does prevent that from happening. Updating the PR now.

> (reg-event-fx
>   :daily-note/delete
>   (fn [{:keys [db]} [_ uid title]]
>     (let [filtered-dn        (filterv #(not= % uid) (:daily-notes/items db)) ;; Filter current date from daily note vec
>           new-db (assoc db :daily-notes/items filtered-dn)]
>       {:fx [[:dispatch [:page/delete uid title]]]
>        :db new-db})))
> ```

**Review by tangjeff0 — APPROVED** [review]
I think `is-timeline-page?` makes more sense than `daily-notes?` from a code perspective, since `daily-notes/items` is persisted even if someone clicks into a different page.

Also, if you try deleting daily notes a few times, it sometimes still fails. I believe this is because we can't guarantee that the `:daily-notes/items` is filtered before `posh` tries to update reactively.

But this works in most cases! Can fix later. 

Some more feedback that turned into a tangent on datascript vs posh, pull vs pull-many, eid vs unique id: https://www.loom.com/share/a1ccd13d60b54a0f82566ba324cf3d37

**Review by tangjeff0 — CHANGES_REQUESTED** [review]
Still getting the alert, but making progress. Please see previous video/comments.

**Pull request closed without merging**